src/models/settlementWindow/facade.js

```diff
@@ -259,17 +255,44 @@ const Facade = {
                 .join('settlementModel AS m', 'm.settlementModelId', 'swc.settlementModelId')
                 .where('tf.settlementWindowId', settlementWindowId)
                 .andWhere('m.settlementGranularityId', Enum.Settlements.SettlementGranularity.NET)
-                .groupBy('swc.settlementWindowContentId', 'pc.participantCurrencyId', 'tp.transferParticipantRoleTypeId', 'tp.ledgerEntryTypeId')
+                .groupBy('swc.settlementWindowContentId', 'pc.participantCurrencyId', 'transferParticipantRoleTypeId', 'ledgerEntryTypeId')
                 .select('swc.settlementWindowContentId', 'pc.participantCurrencyId',
-                  knex.raw('CASE WHEN ?? = NULL THEN ? ELSE ?? END AS transferParticipantRoleTypeId', ['tp.transferParticipantRoleTypeId', 7, 'tp.transferParticipantRoleTypeId']), // @todo 7 = Enum.Accounts.TransferParticipantRoleType.COUNTER_PARTY_FSP requires upgrade of cs-shared
-                  knex.raw('CASE WHEN ?? = NULL THEN ? ELSE ?? END AS ledgerEntryTypeId', ['tp.ledgerEntryTypeId', Enum.Accounts.LedgerEntryType.PRINCIPLE_VALUE, 'tp.ledgerEntryTypeId']),
+                  knex.raw('CASE WHEN ppc.change > 0 THEN ? ELSE ?? END AS transferParticipantRoleTypeId', [ Enum.Accounts.TransferParticipantRoleType.PAYER_DFSP, Enum.Accounts.TransferParticipantRoleType.PAYEE_DFSP]),
+                  knex.raw('? AS ??', [Enum.Accounts.LedgerEntryType.PRINCIPLE_VALUE, 'ledgerEntryTypeId']),
```

Comment:
@oderayi please observe the refactor I did here to avoid use of transferParticipant table and to accommodate FX

Reply:
And also please not, we don't need to use other role types like initiatingFsp...etc. In settlements worl, we can just stick with PAYER_DFSP and PAYEE_DFSP. Otherwise we may need to change other settlement queries for the new role types.

Reply:
This change is cleaner.
LedgerEntryType can also be an Interchange_FEE. Not sure how this change would affect that.
I.e. Could any of the position changes be Interchange_FEE's? 
Here is where they are inserted https://github.com/mojaloop/central-settlement/blob/main/src/models/transferSettlement/facade.js#L40